src/main/java/uk/gov/companieshouse/api/accounts/service/impl/DebtorsService.java

```diff
@@ -0,0 +1,128 @@
+
+        this.repository = repository;
+        this.transformer = transformer;
+        this.smallFullService = smallFullService;
+        this.keyIdGenerator = keyIdGenerator;
+    }
+
+    @Override
+    public ResponseObject<Debtors> create(Debtors rest, Transaction transaction,
+                                          String companyAccountsId, HttpServletRequest request)
+            throws DataException {
+
+        setMetadataOnRestObject(rest, transaction, companyAccountsId);
+
+        DebtorsEntity entity = transformer.transform(rest);
+        entity.setId(generateID(companyAccountsId));
+
+        try {
+
+            repository.insert(entity);
+        } catch (DuplicateKeyException e) {
+
+            LOGGER.errorRequest(request, e, getDebugMap(transaction, companyAccountsId, entity.getId()));
+
+            return new ResponseObject<>(ResponseStatus.DUPLICATE_KEY_ERROR);
+        } catch (MongoException e) {
+
+            DataException dataException = new DataException("Failed to insert " + ResourceName.DEBTORS.getName(), e);
+            LOGGER.errorRequest(request, dataException, getDebugMap(transaction, companyAccountsId, entity.getId()));
+
+            throw dataException;
+        }
+
+        smallFullService
+                .addLink(companyAccountsId, SmallFullLinkType.DEBTORS_NOTE,
+                        getSelfLinkFromDebtors(entity), request);
+
+        return new ResponseObject<>(ResponseStatus.CREATED, rest);
+    }
+
+    @Override
+    public ResponseObject<Debtors> update(Debtors rest, Transaction transaction, String companyAccountId, HttpServletRequest request) throws DataException {
+        return null;
+    }
+
+    @Override
+    public ResponseObject<Debtors> findById(String id, HttpServletRequest request) throws DataException {
+        return null;
+    }
+
+    @Override
+    public String generateID(String companyAccountId) {
+        return keyIdGenerator.generate(companyAccountId + "-" + ResourceName.DEBTORS.getName());
+    }
+
+    private String getSelfLink(Transaction transaction, String companyAccountId) {
+
+        return transaction.getLinks().get(TransactionLinkType.SELF.getLink()) + "/"
+                + ResourceName.COMPANY_ACCOUNT.getName() + "/"
+                + companyAccountId + "/" + ResourceName.SMALL_FULL.getName() + "/"
+                + ResourceName.DEBTORS.getName();
+    }
+
+    public String getSelfLinkFromDebtors(DebtorsEntity entity) {
+
+        return entity.getData().getLinks().get(BasicLinkType.SELF.getLink());
+    }
+
+    private void setLinksOnDebtors(Debtors rest, String link) {
+
+        Map<String, String> map = new HashMap<>();
+        map.put(BasicLinkType.SELF.getLink(), link);
+        rest.setLinks(map);
+    }
+
+    private void setMetadataOnRestObject(Debtors rest, Transaction transaction, String companyAccountsId) {
+
+        setLinksOnDebtors(rest, getSelfLink(transaction, companyAccountsId));
```

Comment:
we could do here something like: 
`rest.setLinks(createSelfLik(transaction, companyAccountsId)` so that it is clear we are setting the rest.setlinks()

Reply:
it would look more like 
` Map<String, String> map = new HashMap<>();
        map.put(BasicLinkType.SELF.getLink(), generateSelfLink(transaction, companyAccountsId));
        rest.setLinks(map);

which is exactly what is in the setLinksOnDebtors method? happy to move this functionality here if it makes more sense?

Reply:
What I meant was that some of the Debtor's fields are being initialized in here , `setMetadataOnRestObject(..)` ; and then, we are calling a different method (`setLinksOnDebtors(..)`) to initialized `rest.links`.  

I would initialize all the fields on the same method for clarity. 

e.g. 

```
private void setMetadataOnRestObject(Debtors rest, Transaction transaction,
    String companyAccountsId) {
    rest.setLinks(createSelfLink(transaction, companyAccountsId));
    rest.setEtag(GenerateEtagUtil.generateEtag());
    rest.setKind(Kind.DEBTORS_NOTE.getValue());
}
```

where: 

```
private Map<String, String> createSelfLink(Transaction transaction, String companyAccountsId) {

    Map<String, String> map = new HashMap<>();
    map.put(BasicLinkType.SELF.getLink(), generateSelfLink(transaction, companyAccountsId));

    return map;
}
```